### DISCUSSION TITLE
Rancher Desktop 1.5.0

### DISCUSSION THREAD

**Discussion by jandubois** [rancher-sandbox/rancher-desktop#2648] [General]
This is the 1.5.0 release of Rancher Desktop, an open source desktop application to bring Kubernetes and container management to Mac, Windows, and Linux. The release contains some new features along with various bug fixes.

### Preferences dialog

Rancher Desktop now has a dedicated preferences dialog containing application, virtual machine, container engine, and Kubernetes settings. Changes are applied only after the "Apply" button is pressed, minimizing the time it takes to switch between configurations.

### Offline operation

Rancher Desktop has been updated to work better in an offline environment. During initial setup a network connection is required to e.g. download the Kubernetes version to be deployed. But if Rancher Desktop is later run offline, then only the already cached versions will be available, and there should no longer be any delays or errors trying to fetch additional information online.

This mode can also be used to setup air-gapped operation. Documentation for this is still [being prepared](https://github.com/rancher-sandbox/rancher-desktop/issues/1358#issuecomment-1180968623).

### Other changes

* The host resolver (DNS) on Windows is no longer considered experimental and is now enabled by default. It helps to resolve hostnames over VPN connections.
 
* The "Port Forwarding" feature now allows the user to pick a specific port instead of choosing one randomly.

## Updated utilities

- `docker` has been upgraded to 20.10.17.
- `docker-compose` has been upgraded to 2.6.1.
- `nerdctl` has been upgraded to 0.22.0.
- `helm` has been upgraded to 3.9.1.
- `trivy` has been upgraded to 0.30.0.

## Known Issuses

* Due to upstream Kubernetes changes the automatic forwarding of NodePorts to the host no longer works for Kubernetes 1.22.10+, 1.23.7+, or any 1.24.1+ release (this also affects all previous Rancher Desktop releases).

* It is not possible on Linux to push images to Docker Hub with `nerdctl`.

Both of these issues will be addressed in a patch release later in August.

You can connect with the developers via:

- The [issue queue](https://github.com/rancher-sandbox/rancher-desktop/issues)
- [Rancher Users Slack](https://slack.rancher.io/) in the #rancher-desktop channel

## Changelog

The [full version changelog, from v1.4.1, can be found using GitHub compare](https://github.com/rancher-sandbox/rancher-desktop/compare/v1.4.1...v1.5.0) and the details of the release can be found in the [v1.5.0 milestone](https://github.com/rancher-sandbox/rancher-desktop/milestone/20?closed=1).

<hr /><em>This discussion was created from the release <a href='https://github.com/rancher-sandbox/rancher-desktop/releases/tag/v1.5.0'>Rancher Desktop 1.5.0</a>.</em>

**Comment by harry-xm** [comment]
Is there an issue on GitHub for "It is not possible to push images to Docker Hub with `nerdctl`." that I can subscribe to?

**Reply by jandubois** [reply]
Yes, it is #2553

**Reply by jandubois** [reply]
I just realized that the release notes didn't say that this is an issue on Linux only; it works correctly on macOS and Windows. I've updated the release notes.

The problem is an incompatibility between `nerdctl` and the `docker-credential-pass` helper. Since Rancher Desktop 1.4.1 it is no longer possible to use `nerdctl` without configuring a credential helper, so this was already broken in 1.4.1 as well.

**Comment by aDisplayName** [comment]
@jandubois 

I have rancher-desktop 1.5.0 installed in my Windows 10 host. I'm assuming it is integrated with the WSL.

After I deployed a simple nginx workflow w/ nodeport service (port 30001) to the kubernetes engine, I'm no longer able to access the nginx services from the browser running under Windows: 
```
curl http://localhost:30001 
```
will simply returns:
```
curl: (7) Failed to connect to localhost port 30001 after 2239 ms: Connection refused
```
while accessing using the node IP of the `rancher-desktop` VM would get the correct response.
```
curl http://192.168.177.137:30001
```

Is the behavior I've observed related to the quote here?
> * Due to upstream Kubernetes changes the automatic forwarding of NodePorts to the host no longer works for Kubernetes 1.22.10+, 1.23.7+, or any 1.24.1+ release (this also affects all previous Rancher Desktop releases).

**Reply by mook-as** [reply]
Yes, that is the referenced issue. ([First PR](https://github.com/rancher-sandbox/rancher-desktop-agent/pull/4) was just posted; after that there will need to be a second PR to actually enable it in RD.)

If you have any suggestions on how we could have better worded the release notes so that it would have been clear for you, they would be very welcome.

**Reply by aDisplayName** [reply]
Thank you!
